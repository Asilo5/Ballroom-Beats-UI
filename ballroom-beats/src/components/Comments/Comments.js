import React, { Component } from 'react';
import { Text, StyleSheet, View, Animated, Dimensions, Easing } from 'react-native';

const { height } = Dimensions.get('window');

const animationY = Math.ceil(height * 0.7);
const negativeEnd = animationY * -1;


class Comments extends Component {

    constructor() {
        super()
        
        this.yAnimation = this.state.position.interpolate({
            inputRange: [negativeEnd, 0],
            outputRange: [animationY, 0] 
          });
   
          this.scaleAnimation = this.yAnimation.interpolate({
             inputRange: [0, 15, 30],
             outputRange: [0, 1.4, 1],
             extrapolate: 'clamp'
          });
    }

    state = {
      position: new Animated.Value(0)
    };

    componentDidMount() {
    //    this.yAnimation = this.state.position.interpolate({
    //      inputRange: [negativeEnd, 0],
    //      outputRange: [animationY, 0] 
    //    });

    //    this.scaleAnimation = this.yAnimation.interpolate({
    //       inputRange: [0, 15, 30],
    //       outputRange: [0, 1.4, 1],
    //       extrapolate: 'clamp'
    //    });

       Animated.timing(this.state.position, {
         duration: 2000,
         toValue: negativeEnd,
         easing: Easing.ease,
         useNativeDriver: true
       }).start();
    }; 

    getStyle = () => {
        return {
            transform: [{ translateY: this.state.position }, { scale: this.scaleAnimation }]
        }
    }

    render() {
        return (
            <Animated.View style={[styles.commentsContainer, this.props.style, this.getStyle()]}>
                <Text style={styles.text} >Got it!</Text>
            </Animated.View>
        );
    }
}

export default Comments;

const styles = StyleSheet.create({
    commentsContainer: {
        position: 'absolute',
        bottom: 30,
        backgroundColor: 'transparent'
    },
    text: {
        color: 'purple',
        fontSize: 30,
        fontWeight: 'bold'
    }
});
