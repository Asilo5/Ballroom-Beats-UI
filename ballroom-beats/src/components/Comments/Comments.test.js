import React from 'react';
import renderer from 'react-test-renderer';
import Comments from './Comments';


describe('Comments', () => {

   it('should match snapshot', () => {
    const comments = renderer.create(<Comments />).toJSON();
    expect(comments).toMatchSnapshot();
   });

});
