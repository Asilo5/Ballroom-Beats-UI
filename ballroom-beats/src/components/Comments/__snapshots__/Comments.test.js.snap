// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comments should match snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "transparent",
      "bottom": 30,
      "position": "absolute",
      "transform": Array [
        Object {
          "translateY": 0,
        },
        Object {
          "scale": 0,
        },
        Object {
          "translateX": 0,
        },
        Object {
          "rotate": "0deg",
        },
      ],
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 30,
        "fontWeight": "bold",
      }
    }
  >
    Doing Great!
  </Text>
</View>
`;
